Library/Homebrew/test/cask/system_command_spec.rb

```diff
@@ -15,46 +15,45 @@
       its("run!.stdout") { is_expected.to eq("123") }
 
       describe "the resulting command line" do
-        it "does not include the given variables" do
+        it "includes the given variables explicitly" do
           expect(Open3)
             .to receive(:popen3)
-            .with(a_hash_including("PATH"), ["env"] * 2, *env_args, {})
+            .with(["env", "env"], "A=1", "B=2", "C=3", "env", *env_args, {})
             .and_call_original
 
           subject.run!
         end
       end
-    end
-
-    describe "given some environment variables and sudo: true" do
-      subject {
-        described_class.new(
-          "env",
-          args: env_args,
-          env: { "A" => "1", "B" => "2", "C" => "3" },
-          must_succeed: true,
-          sudo: true,
-        )
-      }
 
-      describe "the resulting command line" do
-        it "includes the given variables explicitly" do
-          expect(Open3)
-            .to receive(:popen3)
-            .with(an_instance_of(Hash), ["/usr/bin/sudo"] * 2,
-                "-E", a_string_starting_with("PATH="),
-                "A=1", "B=2", "C=3", "--", "env", *env_args, {})
-            .and_wrap_original do |original_popen3, *_, &block|
-              original_popen3.call("/usr/bin/true", &block)
-            end
-
-          subject.run!
+      context "and sudo: true" do
```

Comment:
I feel this part of the code used to be a bit clearer and easier to read correctly when the contexts were not nested.

A condensed example:

```
Something something
  and its result
Something something with sudo
  and its result
```

is easier to read and understand correctly than:

```
Something something
  and its result
  with sudo
    and its result
```

Also, un-nesting won't cost us anything here since the code is duplicated anyway.